lib/identifiers/isbn.rb

```diff
@@ -13,6 +13,7 @@ class ISBN
     /x
     ISBN_10_REGEXP = /
       \b
+      (?<!\p{Pd})       # Do not accidentally match a hyphenated ISBN-13
```

Comment:
Good question: I could make this stricter (don't match an ISBN-10 if it is preceded by a number followed by a space or a dash) but negative lookbehinds are quite restrictive in what you can do.

I'll experiment.